src/mixins/socket.js

```diff
@@ -129,6 +129,18 @@ export default {
                 this.allowLoginDialog = false;
             });
 
+            socket.on("loginRequired", () => {
```

Comment:
Socketio does not have the `at least once` guarantee that this code is actually triggered as far as I read https://socket.io/docs/v4/delivery-guarantees#from-server-to-client

=> users might get stuck without this, or am I seeing this wrong?

Reply:
I see, I wasn't aware of that and none of the other events seem to handle lost message either or am I missing something?

Reply:
Yes, I have not looked into what kind of engeneering effort it would need to get this to retry appropriately.
Idk if this is breaking enough to make this change not viable or if we can assume that if `somebody has established a connection` => `the connection can transmit messages` holds

Reply:
I could add a "worst case" setTimeout falling back to the current handling (i.e. showing the login screen no matter what)

Reply:
websocket runs over TCP, it should have enough delivery guarantees that we can ignore this for now.

Reply:
good point